### PR TITLE
Fix for wso2/product-ei#1243

### PR DESCRIPTION
**Pull request by fazlan-nazeem**
## Purpose
This PR fixes issue https://github.com/wso2/product-ei/issues/1243. This also contains the corresponding UI fixes.

screenshot from 2017-11-02 15-07-35

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.publishevent.ui/src/main/java/org/wso2/carbon/mediator/publishevent/ui/PublishEventMediator.java [hunk 1]
```diff
@@ -137,6 +157,13 @@ public OMElement serialize(OMElement parent) {
 	 * @param elem OMElement to be converted to publishEvent Mediator Object.
 	 */
 	public void build(OMElement elem) {
+		String async = elem.getAttributeValue(ASYNC_Q);
+		this.setIsAsync(async);
+
+		if(Boolean.parseBoolean(async)){
```
Comment (gokul) [inline components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.publishevent.ui/src/main/java/org/wso2/carbon/mediator/publishevent/ui/PublishEventMediator.java]: Minor formatting issue, should be if (...)

Comment (fazlan-nazeem) [reply components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.publishevent.ui/src/main/java/org/wso2/carbon/mediator/publishevent/ui/PublishEventMediator.java]: fixed with e0b1599

File: components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.publishevent.ui/src/main/java/org/wso2/carbon/mediator/publishevent/ui/PublishEventMediator.java [hunk 2]
```diff
@@ -38,6 +40,8 @@
 	public static final QName EVENT_SINK_Q = new QName(XMLConfigConstants.SYNAPSE_NAMESPACE, "eventSink");
 	public static final QName STREAM_NAME_Q = new QName(XMLConfigConstants.SYNAPSE_NAMESPACE, "streamName");
 	public static final QName STREAM_VERSION_Q = new QName(XMLConfigConstants.SYNAPSE_NAMESPACE, "streamVersion");
+	public static final QName ASYNC_Q = new QName( "async");
```
Comment (VijithaEkanayake) [inline components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.publishevent.ui/src/main/java/org/wso2/carbon/mediator/publishevent/ui/PublishEventMediator.java]: Can we fix this formatting issue?

**Comment by fazlan-nazeem** [comment]
@VijithaEkanayake I have fixed according to review comments

**Pull request merged**